src/rpc/net.cpp

```diff
@@ -215,14 +214,19 @@ UniValue addnode(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
 
     std::string strNode = request.params[0].get_str();
+    const bool privileged = request.params[2].isNull() ? true : request.params[2].get_bool();
 
     if (strCommand == "onetry")
     {
         CAddress addr;
-        g_connman->OpenNetworkConnection(addr, false, nullptr, strNode.c_str(), false, false, true);
+        g_connman->OpenNetworkConnection(addr, false, nullptr, strNode.c_str(), false, false, privileged);
         return NullUniValue;
     }
 
+    if (privileged) {
```

Comment:
Should be `if (!request.params[2].isNull()) {`? Furthermore, if the default is `true` then this will always fail.

Reply:
`request.params[2].isNull()` is checked when setting the bool above. But yes, it's inverted here.

Reply:
I mean, it doesn't matter the value, at this point command is not `onetry`, so `privileged` should not be specified.

Reply:
`privileged=true` is required for non-onetry. (Maybe some day we will support `false` too.)

Reply:
The error should be thrown for `(!privileged)` then, it's backwards?